player/magazine.gd

```diff
@@ -0,0 +1,37 @@
+class_name Magazine
+extends Resource
+
+var _max_ammo: int = 7
+var _ammo = _max_ammo
+
+func _init() -> void:
+	ClipSizeUpgrade.new().level_change.connect(_handle_clip_size_upgrade)
```

Comment:
i was confused seeing the clip upgrade constructor called without saving a reference to it. it's clear the previous implementation did the same thing in a different place but it didn't click when i saw it the first time. so we're creating a clip upgrade "listener" and giving it the callback we want to run.

it's weird to conceptualize because that connection exists only because the magazine continues to exist after the fact. i wonder if there would be lingering instances of these connections that run callbacks that change data that's no longer in "view".

Reply:
I would _assume_ that these would get cleared up when the parent elements, the one who created them get cleaned up, should we store a reference just in case? We would not be using them at this time, but maybe for making clear the "ownership"?